File: tests/strings/strings.pas
program test_strings;

var
    firstName: string;
    lastName: string;
    fullName: string;
    greeting: string;

function concat_strings(a: string; b: string): string;
begin
    concat_strings := a + b;
end;

function hello(name: string): string;
begin
    hello := 'Hello, ' + name;
end;

begin
    firstName := 'Nikita';
    lastName := ' Vdovenkov';

    fullName := concat_strings(firstName, lastName);
    greeting := hello(fullName);

    writeln('Full name: ', fullName);
    writeln(greeting);
end.

File: tests/strings/strings.cpp
#include <iostream>
#include <string>

std::string concat_strings(std::string a, std::string b) {
    return (a + b);
}

std::string hello(std::string name) {
    return ("Hello, " + name);
}

int main() {
    std::string firstname;
    std::string lastname;
    std::string fullname;
    std::string greeting;

    firstname = "Nikita";
    lastname = " Vdovenkov";
    fullname = concat_strings(firstname, lastname);
    greeting = hello(fullname);
    std::cout << "Full name: " << " " << fullname << std::endl;
    std::cout << greeting << std::endl;
    return 0;
}